### PR TITLE
feat(translator): Implement BTP CircuitBreaker API

### PR DESCRIPTION
**Pull request by guydc**
**What type of PR is this?**
Feature, implementation of API defined in https://github.com/envoyproxy/gateway/pull/2284

**What this PR does / why we need it**:
Add support for configuring Envoy's [circuit breaking](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/circuit_breaking) settings. 

**Which issue(s) this PR fixes**:
Fixes https://github.com/envoyproxy/gateway/issues/2125

### CODE REVIEW

File: internal/gatewayapi/backendtrafficpolicy.go [hunk 1]
```diff
@@ -482,3 +496,57 @@ func (t *Translator) buildProxyProtocol(policy *egv1a1.BackendTrafficPolicy) *ir
...
+				return nil
+			}
+		}
+	}
+
+	return cb
+}
+
+func setCircuitBreakerPolicyErrorCondition(policy *egv1a1.BackendTrafficPolicy) {
+	message := "Unable to translate Circuit Breaker"
```
Comment (arkodg) [inline internal/gatewayapi/backendtrafficpolicy.go:539]: can we pass an extra `errMsg` to the func to add some more error msg details such as `invalid maxPendingRequests value` etc

**Comment by arkodg** [comment]
the PR is looking really good, thanks for adding an e2e as well !
added a minor comment around err message verbosity in the policy status

File: internal/gatewayapi/backendtrafficpolicy.go [hunk 2]
```diff
@@ -482,3 +496,57 @@ func (t *Translator) buildProxyProtocol(policy *egv1a1.BackendTrafficPolicy) *ir
...
+	status.SetBackendTrafficPolicyCondition(policy,
+		gwv1a2.PolicyConditionAccepted,
+		metav1.ConditionFalse,
+		gwv1a2.PolicyReasonInvalid,
+		message,
+	)
+}
+func int64ToUint32(in int64) (uint32, bool) {
```
Comment (shawnh2) [inline internal/gatewayapi/backendtrafficpolicy.go:547]:
```suggestion

func int64ToUint32(in int64) (uint32, bool) {
```

File: internal/gatewayapi/backendtrafficpolicy.go [hunk 3]
```diff
@@ -482,3 +496,57 @@ func (t *Translator) buildProxyProtocol(policy *egv1a1.BackendTrafficPolicy) *ir
...
+			if ui32, ok := int64ToUint32(*pcb.MaxConnections); ok {
+				cb.MaxConnections = &ui32
+			} else {
+				setCircuitBreakerPolicyErrorCondition(policy)
+				return nil
+			}
+		}
+
+		if pcb.MaxParallelRequests != nil {
+			if ui32, ok := int64ToUint32(*pcb.MaxParallelRequests); ok {
```
Comment (shawnh2) [inline internal/gatewayapi/backendtrafficpolicy.go:517]: can we add some test here? make sure `int64ToUint32` works as expected

**Review by shawnh2 — APPROVED** [review]
some nits, lgtm, thanks

**Review by Xunzhuo — APPROVED** [review]
Thanks, this PR overall looks good ! Pretty clear. Would love to add more e2e tests and a user guide docs in a follow-up.

**Review by cnvergence — APPROVED** [review]

File: internal/gatewayapi/backendtrafficpolicy.go (hunk 1 above)
Comment (guydc) [reply internal/gatewayapi/backendtrafficpolicy.go]: done

File: internal/gatewayapi/backendtrafficpolicy.go (hunk 3 above)
Comment (guydc) [reply internal/gatewayapi/backendtrafficpolicy.go:677]: done

**Pull request merged**